core/src/main/java/org/apache/druid/data/input/impl/DelimitedInputFormat.java

```diff
@@ -98,8 +103,11 @@ protected SeparateValueInputFormat(
         )
     ).getValue();
     this.skipHeaderRows = skipHeaderRows;
-    this.format = format;
-
+    this.format = getFormat(delimiter);
+    Preconditions.checkArgument(
+        delimiter == null || delimiter.length() == 1,
```

Comment:
Since this is created with a "tsv" JSON type and delimiter is annotated with `@Nullable`, `delimiter == null` shouldn't be in this check.

If delimiter is null here, suggest defining a default delimiter (the tab) and setting delimiter to that here, and removing the `delimiter == null` clause in the `else if` in `getFormat`

Reply:
Ah, sorry, I misread the change there in the Preconditions check (it would actually allow null) but I would still suggest clearing the null early and setting it to a default value